### PR TITLE
Core/Units: Prevent backward melee attacks

### PR DESCRIPTION
**Pull request by meji46**
**Changes proposed:**

-  Prevent backward melee attacks from creatures

**Issues addressed:**
None

**Tests performed:**
Builds, tested in-game.
- Possess a creature (.poss)
- Attack another creature in melee
- Face the opposite direction to the attacked creature

**Known issues and TODO list:** (add/remove lines as needed)
None

### CODE REVIEW

**Comment by Ovahlord** [comment]
isn't there a IsInFront helper already which might be better for readability?

**Comment by mdX7** [comment]
> isn't there a IsInFront helper already which might be better for readability?

indeed

```c
bool WorldObject::isInFront(WorldObject const* target, float arc) const
{
    return HasInArc(arc, target);
}
```

**Comment by Shauren** [comment]
Which saves you nothing because the angle value is not equal to default value in `isInFront`, you still call a function on the same object with the same two arguments

**Review by mdX7 — APPROVED** [review]
tested ingame, works fine

File: src/server/game/AI/CoreAI/UnitAI.cpp [hunk 1]
```diff
@@ -68,6 +68,10 @@ void UnitAI::DoMeleeAttackIfReady()
     if (!me->IsWithinMeleeRange(victim))
         return;
 
+    // Check that the victim is in front of the unit
+    if (!me->HasInArc(2 * float(M_PI) / 3, victim))
```
Comment (Nyr97) [inline src/server/game/AI/CoreAI/UnitAI.cpp:72]: Is this 120º degrees?

Comment (mdX7) [reply src/server/game/AI/CoreAI/UnitAI.cpp:72]: yes

**Pull request merged**